src/database.rs

```diff
@@ -134,34 +117,31 @@ fn get_initial_data_from_db(
         }
 
         pb.inc(1);
-        num_rows += 1;
     }
 
-    pb.set_length(num_rows);
+    pb.set_length(pb.position());
     pb.finish();
 
     state_group_map
 }
 
 /// Get any missing state groups from the database
-fn get_missing_from_db(conn: &Connection, missing_sgs: &[i64]) -> BTreeMap<i64, StateGroupEntry> {
-    let stmt = conn
-        .prepare(
+fn get_missing_from_db(client: &mut Client, missing_sgs: &[i64]) -> BTreeMap<i64, StateGroupEntry> {
+    let mut rows = client
+        .query_raw(
             r#"
                 SELECT state_group, prev_state_group
                 FROM state_group_edges
                 WHERE state_group = ANY($1)
             "#,
+            iter::once(&missing_sgs as _),
         )
         .unwrap();
-    let trans = conn.transaction().unwrap();
-
-    let mut rows = stmt.lazy_query(&trans, &[&missing_sgs], 100).unwrap();
```

Comment:
Does `query_raw` do the right thing here? In terms of lazily pulling stuff out of the database?

Reply:
Yes, that was the aim. AFAICT it does, since it returns a `RowIter` instead of a `Vec`.

Reply:
coolies